### DISCUSSION TITLE
Support multi-threaded download

### DISCUSSION THREAD

**Discussion by 2284787774** [Dineshkarthik/telegram_media_downloader#171] [Q&A]
Ability to set the number of simultaneous downloads

**Comment by Dineshkarthik** [comment]
Currently the script process/downloads messages concurrently using asyncio. The number of messages processed concurrenlty is set here [media_downloader.py#L353](https://github.com/Dineshkarthik/telegram_media_downloader/blob/master/media_downloader.py#L353) - pagination_limit.
Currently a maximum of 100 messages are processed concurrently and the concurrent downloads will be based on the available system resources.
